File: java/src/main/java/com/workshop/architecture/fitness/infrastructure/InMemoryEmailService.java
package com.workshop.architecture.fitness.infrastructure;

import java.util.ArrayList;
import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class InMemoryEmailService {

    private final List<String> sentEmails = new ArrayList<>();

    public void send(String emailContent) {
        sentEmails.add(emailContent);
    }

    public List<String> sentEmails() {
        return List.copyOf(sentEmails);
    }

    public void clear() {
        sentEmails.clear();
    }
}

File: python/src/workshop_api/fitness/managing_memberships/shared/in_memory_email_service.py
class InMemoryEmailService:
    def __init__(self) -> None:
        self._sent_emails: list[str] = []

    def send(self, email_content: str) -> None:
        self._sent_emails.append(email_content)

    def sent_emails(self) -> list[str]:
        return list(self._sent_emails)

    def clear(self) -> None:
        self._sent_emails.clear()


email_service = InMemoryEmailService()
